Lesson18/main.js

```diff
@@ -0,0 +1,218 @@
+  slideItem.appendChild(slideImg);
+  return slideItem;
+}
+
+const createArrowButtons = () => {
+  const arrowBtnWrapper = createElementWithClassName("div", "arrowBtn-wrapper");
+  const arrowDirections = ["previous", "next"];
+  arrowDirections.forEach((arrowDirection) => {
+    const button = createElementWithClassName("button", `arrowBtn --${arrowDirection}`);
+    button.id = `js-${arrowDirection}Btn`;
+    button.value = arrowDirection;
+    /**
+     * Set previous attribute to disabled by default.
+     */
+    button.value === "previous" && button.setAttribute("disabled", true);
+    arrowBtnWrapper.appendChild(button).appendChild(document.createElement("span"));
+  });
+  return arrowBtnWrapper;
+}
+
+const createIndicator = (imageLength) => {
+  const ul = createElementWithClassName("ul", "indicator-list")
+  for (let i = 0; i < imageLength; i++) {
+    const li = createElementWithClassName("li", "indicator-item");
+    li.setAttribute("data-num", i);
+    /**
+    * Set the is-selected class to the first element by default 
+    */
+    i === 0 && li.classList.add("is-selected");
+    ul.appendChild(li);
+  }
+  return ul;
+}
+
+const createCounterOfSlide = (data) => {
+  const counterWrapper = createElementWithClassName("div", "counter");
+  const orderOfDisplayedItem = createElementWithClassName("span", "current-number");
+  const totalSlideItem = createElementWithClassName("span", "total-number");
+  totalSlideItem.textContent = data.length;
+  /**
+   * Add 1 to the currently displayed index and display the count.
+   */
+  orderOfDisplayedItem.textContent = findIndexOfDisplayedItem() + 1;
+  counterWrapper.appendChild(orderOfDisplayedItem).insertAdjacentHTML("afterend", "/");
+  counterWrapper.appendChild(totalSlideItem);
+  return counterWrapper;
+}
+
+const findIndexOfDisplayedItem = () => {
+  const slideItemArray = [...document.querySelectorAll(".slide-item")];
+  return slideItemArray.findIndex(el => el.classList.contains("is-displaying"));
+}
+
+const findIndexOfSelectedIndicator = () => {
+  const indicators = [...document.querySelectorAll(".indicator-item")];
+  return indicators.findIndex((el) => el.classList.contains("is-selected"));
+}
+
+const toggleTheDisabled = () => {
+  const lastSlideItem = slideList.lastElementChild;
+  const firstSlideItem = slideList.firstElementChild;
+  const displayingEl = document.querySelector(".is-displaying");
+  const disabledEl = document.querySelector("[disabled]");
+  disabledEl && disabledEl.removeAttribute("disabled")
+  if (displayingEl === lastSlideItem) {
+    document.getElementById("js-nextBtn").setAttribute("disabled", true);
+  }
+  if (displayingEl === firstSlideItem) {
+    document.getElementById("js-previousBtn").setAttribute("disabled", true);
+  }
+}
+
+const updateOfCounter = () => {
+  const displayedItemIndex = findIndexOfDisplayedItem();
+  document.querySelector(".current-number").textContent = displayedItemIndex + 1;
+}
+
+const switchIndicator = (target) => {
```

Comment:
```suggestion
const switchIndicator = (targetIndex) => {
```

Reply:
Thank you for your review.
We have made the corrections.
Please check it out.

5374c0f